apps/cowswap-frontend/src/modules/ethFlow/services/ethFlow/index.ts

```diff
@@ -81,21 +78,56 @@ export async function ethFlow({
       )
     }
 
-    logTradeFlow('ETH FLOW', 'STEP 4: sign order')
-    const { order, txReceipt } = await signEthFlowOrderStep(
+    logTradeFlow('ETH FLOW', 'STEP 3: sign order')
+
+    const {
       orderId,
-      orderParams,
-      contract,
-      addInFlightOrderId,
-      tradeContext,
-    ).finally(() => {
-      callbacks.closeModals()
+      txHash,
+      signature,
+      signingScheme,
+      orderToSign: unsignedOrder,
+    } = await wrapErrorInOperatorError(() =>
+      tradeQuote
+        .quote!.postSwapOrderFromQuote({
```

Comment:
I'm not found of non-null assertions, but I don't want to block this change.
Please do come back to it later and throw a explicit error instead.

Reply:
Thanks!
Fixed that: https://github.com/cowprotocol/cowswap/pull/5549/commits/e901c98458819c074859ebef7bdfbaffe4ce27e8